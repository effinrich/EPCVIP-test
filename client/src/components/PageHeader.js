import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading } from 'grommet'

const PageHeader = ({ name, headerElement }) => (
  <Box
    flex={false}
    margin={{ bottom: 'medium' }}
    justify="between"
    align="center"
    direction="row"
    border={{ side: 'bottom', color: 'light-4' }}
  >
    <Heading level={1} size="xsmall" margin={{ top: 'xsmall' }}>
      {name}
    </Heading>
    {headerElement && <Box alignSelf="center">{headerElement}</Box>}
  </Box>
)

PageHeader.propTypes = {
  name: PropTypes.string,
  headerElement: PropTypes.node
}

export default PageHeader
