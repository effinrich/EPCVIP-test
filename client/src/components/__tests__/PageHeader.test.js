import React from 'react'

import { render } from 'utils/testUtils/helperRtl'

import PageHeader from '../PageHeader'

describe('PageHeader Component', () => {
  it('should render without crashing', () => {
    const { container } = render(<PageHeader />)
    expect(container).toMatchSnapshot()
  })

  it('should render child element', () => {
    const { container } = render(<PageHeader />)
    expect(container.firstChild).toBeDefined()
  })
})
