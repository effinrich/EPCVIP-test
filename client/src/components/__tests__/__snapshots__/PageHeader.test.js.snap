// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render without crashing 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 fTeJqU"
  >
    <h1
      class="StyledHeading-sc-1rdh4aw-0 dtaByy"
    />
  </div>
</div>
`;
